packages/next/client/link.tsx

```diff
@@ -459,14 +456,14 @@ const Link = React.forwardRef<HTMLAnchorElement, LinkPropsReal>(
     }, [as, href, isVisible, locale, p, router])
 
     const childProps: {
-      onTouchStart: React.TouchEventHandler
-      onMouseEnter: React.MouseEventHandler
-      onClick: React.MouseEventHandler
+      onTouchStart: React.TouchEventHandler<HTMLAnchorElement>
+      onMouseEnter: React.MouseEventHandler<HTMLAnchorElement>
+      onClick: React.MouseEventHandler<HTMLAnchorElement>
       href?: string
       ref?: any
     } = {
       ref: setRef,
-      onClick: (e: React.MouseEvent) => {
+      onClick: (e) => {
```

Comment:
The type of `e` can be inferred from above, so implicit typing is no longer required here.